From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 23 Feb 2015 21:37:52 +0000
Subject: PCI: quirks: Fix backport of quirk_io()

Commit 06cf35f903aa ('PCI: Handle read-only BARs on AMD CS553x
devices') added the function quirk_io() which calls
pcibios_bus_to_resource().

Prior to Linux 3.14, pcibios_bus_to_resource() takes a pointer to
struct pci_dev and looks up the device's bus itself, so we need
to pass dev not dev->bus.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -409,7 +409,7 @@ static void quirk_io(struct pci_dev *dev
 	/* Convert from PCI bus to resource space */
 	bus_region.start = region;
 	bus_region.end = region + size - 1;
-	pcibios_bus_to_resource(dev->bus, res, &bus_region);
+	pcibios_bus_to_resource(dev, res, &bus_region);
 
 	dev_info(&dev->dev, FW_BUG "%s quirk: reg 0x%x: %pR\n",
 		 name, PCI_BASE_ADDRESS_0 + (pos << 2), res);
